os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -279,28 +277,22 @@ static inline int imxrt_bulkerase(struct imxrt_dev_s *priv, unsigned long arg)
 	/* Deselect the FLASH */
 
 	SPI_SELECT(priv->dev, SPIDEV_FLASH, false);
+#endif
 	
-	size_t sectors_erased = 0;
-	size_t startsector = 8;
-//	fdbg("startsector: %d nsectors: %d\n", (long)startsector, (int)priv->nsectors);
-	fdbg("startsector: %d nsectors: %d\n", (long)startsector, 2);
+	off_t sectors_erased = 0;
+	fvdbg("priv: %p, startblock: %d, startaddr: %x, nsectors: %d\n", priv, startblock, startblock >> priv->pageshift, nsectors);
 
 	/* Lock access to the SPI bus until we complete the erase */
 
-//	while (sectors_erased < priv->nsectors) {
-	while (sectors_erased < 2) {
+	while (sectors_erased < priv->nsectors) {
 		/* Not using sub-sector erase.  Erase each full sector */
 
-		imxrt_sectorerase(priv, startsector + sectors_erased);
+		imxrt_sectorerase(priv, sectors_erased);
 		sectors_erased++;
 	}
+	fvdbg("erased sectors = %d\n", sectors_erased);
 
-	fdbg("Return = erased sectors = %d\n", sectors_erased);
-	return sectors_erased;
-	#endif
-
-	fdbg("Return: OK\n");
-	return OK;
+	return sectors_erased == priv->nsectors ? OK : ERROR;
```

Comment:
This is bulkerase, you should check erased block number, if it is same as total block, then OK.
Otherwise -EIO

Reply:
Please refer to fs_ioctl.c, you can realize how ioctl controls return value of each ops.

Reply:
Modified as suggested.